dyn_em/module_big_step_utilities_em.F

```diff
@@ -2545,154 +2553,210 @@ SUBROUTINE w_damp( rw_tend, max_vert_cfl,max_horiz_cfl, &
    max_vert_cfl = 0.
    max_horiz_cfl = 0.
    total = 0
 
+! LJW BEGIN
+
+   w_crit_cfl = config_flags%w_crit_cfl
+
+   IF( print_flag ) THEN
+     write(wrf_err_message,*) '----------------------------------------'
+     CALL wrf_debug( 0, wrf_err_message )
+     WRITE(temp,*) 'W-FLAGGING BEGINS AT W-COURANT NUMBER = ',w_flag_cfl
+     CALL wrf_debug ( 0 , TRIM(temp) )
+     WRITE(temp,*) 'W-DAMPING  BEGINS AT W-COURANT NUMBER = ',w_crit_cfl
+     CALL wrf_debug ( 0 , TRIM(temp) )
+     write(wrf_err_message,*) '----------------------------------------'
+     CALL wrf_debug( 0, wrf_err_message )
+     print_flag = .false.
+   ENDIF
+
+! LJW END
+
    IF(config_flags%polar ) then
      msfxffl = 1.0/COS(config_flags%fft_filter_lat*degrad)
    END IF
 
+! LJW BEGIN
+
+! There is a lot of redundant code due to wanting information whether w_damping is turned on or not.
+! I decided that instead of multiple blocks of nearly identical code - I would simple have a constant of
+! (0.0,1.0) if damping was on or off to turn things on and off and then execute the same code.
+! Since w_damping is almost always on for NWP application, the cost would be the same.
+
    IF ( config_flags%w_damping == 1 ) THEN
-#ifdef OPTIMIZE_CFL_TEST
+
+     w_damp_on = 1.0
+
+   ELSE
+ 
+     w_damp_on = 0.0
+
+   ENDIF
+
+!#ifdef OPTIMIZE_CFL_TEST
 ! 20121025, L. Meadows vector optimization does not include special case for Cassini
-     IF(config_flags%polar ) then
-       CALL wrf_error_fatal('module_big_step_utilities_em.F: -DOPTIMIZE_CFL_TEST option does not support global domains')
-     END IF
-#endif
+!    IF(config_flags%polar ) then
+!      CALL wrf_error_fatal('module_big_step_utilities_em.F: -DOPTIMIZE_CFL_TEST option does not support global domains')
+!    END IF
+!endif
 
-     DO j = jts,jtf
+   DO j = jts,jtf
+     DO k = 2,kde-1
+       DO i = its,itf
 
-     DO k = 2, kde-1
-     DO i = its,itf
-#if 1
-# ifdef OPTIMIZE_CFL_TEST
-! L. Meadows, Intel, MIC optimization, 20121025
-        msfuxt = msfux(i,j)
-        vert_cfl = abs(ww(i,k,j)/(c1f(k)*mut(i,j)+c2f(k))*rdnw(k)*dt)
-        IF ( vert_cfl > max_vert_cfl ) THEN
-           max_vert_cfl = vert_cfl
-        ENDIF
-# else
-        IF(config_flags%polar ) then
-           msfuxt = MIN(msfux(i,j), msfxffl)
-        ELSE
-           msfuxt = msfux(i,j)
-        END IF
-        vert_cfl = abs(ww(i,k,j)/(c1f(k)*mut(i,j)+c2f(k))*rdnw(k)*dt)
+! LJW BEGIN COMMENT OUT
+! I doubt whether this old code is needed with newer compilers - so I commented it all out
 
-        IF ( vert_cfl > max_vert_cfl ) THEN
-           max_vert_cfl = vert_cfl ; maxi = i ; maxj = j ; maxk = k
-           maxdub = w(i,k,j) ; maxdeta = -1./rdnw(k)
-        ENDIF
-# endif
-        
-        horiz_cfl = max( abs(u(i,k,j) * rdx * msfuxt * dt),                          &
-             abs(v(i,k,j) * rdy * msfvy(i,j) * dt) )
-        if (horiz_cfl > max_horiz_cfl) then
-           max_horiz_cfl = horiz_cfl
-        endif
+!#if 1
```

Comment:
@weiwangncar @dudhia 
Folks, 
Do we want to comment this code out (as Lou has done), or just toss it out completely. My vote is remove it.